### PR TITLE
Fix const generic arguments not displaying in types mismatch diagnostic

### LINKED ISSUE #61395: Const-generic arguments are not printed in error messages
This code ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=71becd94ee21012d9ce69a2d8744d964)):

```rust
#![feature(const_generics)]

struct Generic<const C: usize>;

fn main() {
    let _: Generic<0> = Generic::<1>;
}
```

results in

```
error[E0308]: mismatched types
 --> src/main.rs:6:25
  |
6 |     let _: Generic<0> = Generic::<1>;
  |                         ^^^^^^^^^^^^ expected `0usize`, found `1usize`
  |
  = note: expected type `Generic<>`
             found type `Generic<>`
```

with missing const arguments in the `note: ...` at the bottom.

### PR DESCRIPTION
**Pull request by BenLewis-Seequent**
Fixes #61395

### CODE REVIEW

File: src/librustc/infer/error_reporting/mod.rs [hunk 1]
```diff
@@ -994,6 +996,21 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                         self.push_comma(&mut values.0, &mut values.1, len, i);
                     }
 
+                    // Do the same for const arguments, if they are equal, do not highlight and
+                    // elide them from the output.
+                    let const_arguments = sub1.consts().zip(sub2.consts());
+                    for (i, (ca1, ca2)) in const_arguments.enumerate() {
+                        let i = i + consts_offset;
+                        if ca1 == ca2 {
+                            values.0.push_normal("_");
```
Comment (fbstj) [inline src/librustc/infer/error_reporting/mod.rs]: are there tests for this path? where some other error happens when you have `T<"a", "b"> != T<"A","b">` for example (or even for when you have const generics that are equal but other things wrong?

Comment (skinny121) [reply src/librustc/infer/error_reporting/mod.rs]: I have added tests for this.

**Comment by varkor** [comment]
Thanks, this looks great!

@bors r+ rollup

**Pull request merged**